Chessington.GameEngine/Pieces/MoveVector.cs

```diff
@@ -28,17 +28,58 @@ public static IEnumerable<Square> FindAvailableMovesFromVectors(IEnumerable<Move
 
             var possibleMovesInBoard = RemoveInvalidSquares(possibleMoves);
 
-            // Remove current location
-            var availableLocations = possibleMovesInBoard.Where(square => square != location);
+            // Remove locations containing friendly pieces
+            var availableLocations = RemoveFriendlyOccupiedSquares(possibleMovesInBoard, board);
 
             return availableLocations;
```

Comment:
This can be merged with the variable declaration above 🙂

Reply:
Yep, sorted in 51a4e8d 🙂